src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx

```diff
@@ -32,23 +34,43 @@ interface Props {
  * @returns {React.FC<Props>} - The rendered confirmation form.
  */
 export const ConfirmationForm: React.FC<Props> = ({ environment, type }) => {
-  const { commandResolver } = useContext(DependencyContext);
   const { closeModal } = useContext(ModalContext);
   const navigateTo = useNavigateTo();
+  const client = useQueryClient();
 
   const [candidateEnv, setCandidateEnv] = useState("");
   const [errorMessage, setErrorMessage] = useState<null | string>(null);
   const [isBusy, setIsBusy] = useState(false);
   const validated = environment.name === candidateEnv ? "success" : "default";
 
   const redirectToHome = () => navigateTo("Home", undefined);
-  const deleteTrigger = commandResolver.useGetTrigger<"DeleteEnvironment">({
-    kind: "DeleteEnvironment",
-    id: environment.id,
+
+  const deleteEnv = useDeleteEnvironment(environment.id, {
+    onSuccess: () => {
+      //reset the queries removes the cache which improves the ux when navigating back to the environments page
```

Comment:
What's the link with UX? Not sure I understand the comment

Reply:
we wouldn't show loading state when there is cache so fro split second there would be X envs and after deletion X-1